pkg/vmprovider/providers/vsphere/contentlibrary/content_library_utils.go

```diff
@@ -103,6 +103,41 @@ func LibItemToVirtualMachineImage(
 	return image
 }
 
+// UpdateVmiWithOvfEnvelope updates the given vmi object with the content of given OVF envelope.
+func UpdateVmiWithOvfEnvelope(vmi client.Object, ovfEnvelope ovf.Envelope) {
+	var spec *v1alpha1.VirtualMachineImageSpec
+	var status *v1alpha1.VirtualMachineImageStatus
+	switch vmi := vmi.(type) {
+	case *v1alpha1.VirtualMachineImage:
+		spec = &vmi.Spec
+		status = &vmi.Status
+	case *v1alpha1.ClusterVirtualMachineImage:
+		spec = &vmi.Spec
```

Comment:
what is the point of switching cases when setting it to the same value? Are you trying to guard against other types than vmi/cvmi?

Reply:
Because he's casting it, and if you combine them into a single case, it won't let you access the fields of that type.

Reply:
What Andrew said :) Btw, this is from the existing code and I am just moving it to a separate function in the `content_library_utils.go` file (as it doesn't contain any dependency on the client).